src/main/java/com/nylas/Notification.java

```diff
@@ -150,6 +150,13 @@ public String toString() {
 	public static class Attributes {
 		private String received_date;
 		private String thread_id;
+
+		// message.bounced specific fields
+		private String message_id;
+		private String original_message_id;
+		private String original_thread_id;
+		private String bounce_reason;
+		private String bounce_diagnostic;
```

Comment:
I don't think we want to add specific named fields for every notification type into the same Attributes object.  It looks like the two existing fields are specific to the `message.created` notification, since that was the only one I found at the time that had any attributes.  Now I can also see a `job.successful` notification with attributes `action` and `job_status_id` as well as this new `message.bounced` with the given 5 fields.  There may be others too.

Unfortunately, the alternatives do not seem as simple to accomplish.  I think it would be good to change the Attributes class to be an interface with implementations specific to each notification type that provides any attributes.  But the JSON serialization becomes somewhat tricky because the best discriminator field seems to be `Delta.type` rather than a direct field on the Attributes JSON.  (Compare to Event.When where the server includes an `object` field directly in the json that determines its type).  So to perform the deserialization using the Moshi library will require some custom work.  One possible example is at https://github.com/square/moshi/issues/813#issuecomment-464808616 but I haven't fully groked it yet.

Or maybe you have another idea?  What do you think?